File: frontend/src/types/issue.ts
import type { PullRequest } from 'types/pullRequest'
import type { RepositoryDetails, User } from 'types/user'

export type Issue = {
  author?: User
  body?: string
  createdAt: number
  hint?: string
  labels?: string[]
  number?: string | number
  objectID?: string
  organizationName?: string | null
  projectName?: string
  projectUrl?: string
  pullRequests?: PullRequest[]
  repository?: RepositoryDetails
  repositoryLanguages?: string[]
  repositoryName?: string | null
  state?: string
  summary?: string
  title: string
  updatedAt?: number
  url: string
}

File: backend/apps/graphql/types/github/issue.py
from apps.github.models.issue import Issue
from apps.github.models.user import User
from graphene_django import DjangoObjectType

class IssueType(DjangoObjectType):
    class Meta:
        model = Issue
        fields = ('number', 'title', 'created_at', 'comments_count', 'author', 'state')